### PR TITLE
Commerce mock test refactoring

### PR DESCRIPTION
**Pull request by pbochynski**

### CODE REVIEW

**Review by aerfio — CHANGES_REQUESTED** [review]
I was not able to test it locally due to issues with watch function

File: tests/fast-integration/test/commerce-mock.js [hunk 1]
```diff
@@ -55,169 +57,113 @@ describe("Commerce Mock tests", function () {
...
+  it("commerce mock should connect to Kyma", async function () {
+    this.retries(4)
+    const tokenRequest = {
+      apiVersion: 'applicationconnector.kyma-project.io/v1alpha1',
+      kind: 'TokenRequest',
+      metadata: { name: 'commerce', namespace: 'default' }
+    }
+    const path = `/apis/applicationconnector.kyma-project.io/v1alpha1/namespaces/default/tokenrequests`
+
+    await k8sDynamicApi.delete(tokenRequest).catch(() => { })// Ignore delete error
```
[the remark this answers is not in the document]
Comment (pbochynski) [reply tests/fast-integration/test/commerce-mock.js:121]: The token can be used only once and is valid only for 5 minutes. Retry with the same token doesn't make sense - you need new tokenrequest resource for each retry.

File: tests/fast-integration/test/commerce-mock.js [hunk 2]
```diff
@@ -27,18 +27,20 @@ const kc = new k8s.KubeConfig();
...
 
 const k8sDynamicApi = kc.makeApiClient(k8s.KubernetesObjectApi);
-const k8sCRDApi = kc.makeApiClient(k8s.CustomObjectsApi);
 const k8sAppsApi = kc.makeApiClient(k8s.AppsV1Api);
+const watch = new k8s.Watch(kc);
 
 const commerceObjs = k8s.loadAllYaml(commerceMockYaml);
 const mocksNamespaceObj = k8s.loadYaml(mocksNamespaceYaml);
+let mockHost;
+let host;
```
[the remark this answers is not in the document]
Comment (pbochynski) [reply tests/fast-integration/test/commerce-mock.js]: done

File: tests/fast-integration/test/commerce-mock.js [hunk 3]
```diff
@@ -27,18 +27,20 @@ const kc = new k8s.KubeConfig();
...
 
 describe("Commerce Mock tests", function () {
   this.timeout(10 * 60 * 1000);
+  this.slow(1000);
```
[the remark this answers is not in the document]
Comment (pbochynski) [reply tests/fast-integration/test/commerce-mock.js]: Mocha reports slow tests with red color. I set it to 5000.

File: tests/fast-integration/test/commerce-mock.js [hunk 4]
```diff
@@ -55,169 +57,113 @@ describe("Commerce Mock tests", function () {
...
   });
 
-  // TODO: check if we can split this one big "it" into several smaller ones
-  // mocha should preserve the order of test inside "describe" block, but I'm not sure
-  it("should pass with 😄", async function () {
-    console.log("Creating mocks namespace...");
-    await k8sDynamicApi.create(mocksNamespaceObj).catch(expectNoK8sErr); // we can extract namespace creation into seperate step and ignore AlreadyExists error
+  it("mocks namespace should be created or fail with reason AlreadyExists ", async function () {
+    await k8sDynamicApi.create(mocksNamespaceObj).catch((e) => {
+      expect(e.body.reason).to.be.equal('AlreadyExists')
```
[the remark this answers is not in the document]
Comment (pbochynski) [reply tests/fast-integration/test/commerce-mock.js]: done

File: tests/fast-integration/test/commerce-mock.js [hunk 5]
```diff
@@ -55,169 +57,113 @@ describe("Commerce Mock tests", function () {
...
-
-    await registerApis(
-      `https://${mockHost}/local/apis/Commerce%20Webservices/register`,
-      mockHost
+      {
+        headers: {
+          "content-type": "application/json",
+        },
+      }
     );
```
[the remark this answers is not in the document]
Comment (pbochynski) [reply tests/fast-integration/test/commerce-mock.js]: done

File: tests/fast-integration/test/commerce-mock.js [hunk 6]
```diff
@@ -55,169 +57,113 @@ describe("Commerce Mock tests", function () {
...
-      elem.name.includes("Commerce Webservices")
-    ).id;
-    expect(commerceWebservicesID).not.to.be.empty;
-
-    await registerApis(
-      `https://${mockHost}/local/apis/Events/register`,
-      mockHost
-    );
+  it("commerce mock should register Commerce Webservices API and Events", async function () {
+    this.retries(10);
```
[the remark this answers is not in the document]
Comment (pbochynski) [reply tests/fast-integration/test/commerce-mock.js]: done

File: tests/fast-integration/test/commerce-mock.js [hunk 7]
```diff
@@ -55,169 +57,113 @@ describe("Commerce Mock tests", function () {
...
-    ).catch(expectNoK8sErr);
-
-    expect(eventsServiceClass.body).to.have.nested.property(
-      "spec.externalName"
-    );
-    const eventsSCExternalName = eventsServiceClass.body.spec.externalName;
+    ).externalName;
 
-    const serviceCatalogObjs = k8s.loadAllYaml(
+    const serviceCatalogObjs = await k8s.loadAllYaml(
```
[the remark this answers is not in the document]
Comment (pbochynski) [reply tests/fast-integration/test/commerce-mock.js]: right

File: tests/fast-integration/test/commerce-mock.js [hunk 8]
```diff
@@ -239,7 +185,7 @@ async function sendEventAndCheckResponse(mockHost, host) {
             "content-type": "application/json",
           },
         }
-      );
+      ).catch(e => console.dir({ status: e.response.status, data: e.response.data }));
```
[the remark this answers is not in the document]
Comment (pbochynski) [reply tests/fast-integration/test/commerce-mock.js]: good point

**Review by aerfio — APPROVED** [review]

**Pull request merged**